conda/recipes/libcucim/meta.yaml

```diff
@@ -17,7 +17,6 @@ build:
   number: {{ GIT_DESCRIBE_NUMBER }}
   string: cuda{{ cuda_major }}_{{ date_string }}_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
   ignore_run_exports:
-    - libwebp-base
```

Comment:
*(choosing a part of the diff to comment on just so we can have a threaded conversation that could eventually be marked "resolved")*

---

I can help explain this error in the `conda-cpp-build` jobs:

```text
conda.CondaMultiError: This transaction has incompatible packages due to a shared path.
  packages: conda-forge/linux-64::xorg-renderproto-0.11.1-hb9d3cd8_1003, conda-forge/linux-64::xorg-xorgproto-2024.1-hb9d3cd8_1
  path: 'include/X11/extensions/render.h'
```

([build link](https://github.com/rapidsai/cucim/actions/runs/11058127444/job/30723517800?pr=785))

First, it happens shortly after this log line:

```text
BUILD START: ['libcucim-24.10.00a21-cuda11_240926_g303981a_21.tar.bz2']
```

That tells you that this is happening in the part where `conda-build` is setting up the **build** environment, so shouldn't be related to `cucim`'s runtime-only dependencies.

This error message means that both `xorg-renderproto` and `xorg-xorgproto` both include X11 headers with the same filepaths.

I looked at the most recent nightly build of `cucim`, from a few hours ago. It succeeded ([build link](https://github.com/rapidsai/cucim/actions/runs/11045891755/job/30684309021)), and installed these versions of the `xorg-*` libraries:

```text
    xorg-kbproto:              1.0.7-hb9d3cd8_1003  conda-forge
    xorg-libice:               1.1.1-hd590300_0     conda-forge
    xorg-libsm:                1.2.4-h7391055_0     conda-forge
    xorg-libx11:               1.8.9-hb711507_1     conda-forge
    xorg-libxau:               1.0.11-hb9d3cd8_1    conda-forge
    xorg-libxdmcp:             1.1.3-hb9d3cd8_1     conda-forge
    xorg-libxext:              1.3.4-h0b41bf4_2     conda-forge
    xorg-libxrender:           0.9.11-hd590300_0    conda-forge
    xorg-renderproto:          0.11.1-hb9d3cd8_1003 conda-forge
    xorg-xextproto:            7.3.0-hb9d3cd8_1004  conda-forge
    xorg-xproto:               7.0.31-hb9d3cd8_1008 conda-forge
```

The failing jobs here are getting these versions:

```text
    xorg-libice:               1.1.1-hd590300_0     conda-forge
    xorg-libsm:                1.2.4-h7391055_0     conda-forge
    xorg-libx11:               1.8.10-h4f16b4b_0    conda-forge
    xorg-libxau:               1.0.11-hb9d3cd8_1    conda-forge
    xorg-libxdmcp:             1.1.3-hb9d3cd8_1     conda-forge
    xorg-libxext:              1.3.4-h0b41bf4_2     conda-forge
    xorg-libxrender:           0.9.11-hd590300_0    conda-forge
    xorg-renderproto:          0.11.1-hb9d3cd8_1003 conda-forge
    xorg-xextproto:            7.3.0-hb9d3cd8_1004  conda-forge
    xorg-xorgproto:            2024.1-hb9d3cd8_1    conda-forge
```

Notice the difference:

* nightly: `xorg-xproto`, but no `xorg-xorgproto`
* CI here: no `xorg-xproto`, but has `xorg-xorgproto`

I'm not familiar with those libraries, so not sure why that's happening. Will keep looking.

Reply:
I found a small reproducible example. On an x86_64 machine with CUDA 12.2, ran the following:

```shell
docker run \
    --rm \
    -it rapidsai/ci-conda:latest \
    bash

conda config --set path_conflict prevent

conda install \
    --override-channels \
    -c conda-forge \
        openslide
```

That's sufficient to reproduce the errors about incompatible paths with these `xorg-*` libraries!

Here's the full set of what `conda` attempts to install as a result of solving that:

```text
  cairo              conda-forge/linux-64::cairo-1.18.0-hebfffa5_3 
  expat              conda-forge/linux-64::expat-2.6.3-h5888daf_0 
  font-ttf-dejavu-s~ conda-forge/noarch::font-ttf-dejavu-sans-mono-2.37-hab24e00_0 
  font-ttf-inconsol~ conda-forge/noarch::font-ttf-inconsolata-3.000-h77eed37_0 
  font-ttf-source-c~ conda-forge/noarch::font-ttf-source-code-pro-2.038-h77eed37_0 
  font-ttf-ubuntu    conda-forge/noarch::font-ttf-ubuntu-0.83-h77eed37_2 
  fontconfig         conda-forge/linux-64::fontconfig-2.14.2-h14ed4e7_0 
  fonts-conda-ecosy~ conda-forge/noarch::fonts-conda-ecosystem-1-0 
  fonts-conda-forge  conda-forge/noarch::fonts-conda-forge-1-0 
  freetype           conda-forge/linux-64::freetype-2.12.1-h267a509_2 
  gdk-pixbuf         conda-forge/linux-64::gdk-pixbuf-2.42.12-hb9ae30d_0 
  lerc               conda-forge/linux-64::lerc-4.0.0-h27087fc_0 
  libdeflate         conda-forge/linux-64::libdeflate-1.21-h4bc722e_0 
  libdicom           conda-forge/linux-64::libdicom-1.0.5-hd590300_1 
  libglib            conda-forge/linux-64::libglib-2.80.5-h2ff4ddf_0 
  libjpeg-turbo      conda-forge/linux-64::libjpeg-turbo-3.0.0-hd590300_1 
  libpng             conda-forge/linux-64::libpng-1.6.44-hadc24fc_0 
  libtiff            conda-forge/linux-64::libtiff-4.7.0-h6565414_0 
  libwebp-base       conda-forge/linux-64::libwebp-base-1.4.0-hd590300_0 
  libxcb             conda-forge/linux-64::libxcb-1.17.0-h8a09558_0 
  openjpeg           conda-forge/linux-64::openjpeg-2.5.2-h488ebb8_0 
  openslide          conda-forge/linux-64::openslide-4.0.0-h58ba908_1 
  pixman             conda-forge/linux-64::pixman-0.43.2-h59595ed_0 
  pthread-stubs      conda-forge/linux-64::pthread-stubs-0.4-hb9d3cd8_1002 
  xorg-libice        conda-forge/linux-64::xorg-libice-1.1.1-hd590300_0 
  xorg-libsm         conda-forge/linux-64::xorg-libsm-1.2.4-h7391055_0 
  xorg-libx11        conda-forge/linux-64::xorg-libx11-1.8.10-h4f16b4b_0 
  xorg-libxau        conda-forge/linux-64::xorg-libxau-1.0.11-hb9d3cd8_1 
  xorg-libxdmcp      conda-forge/linux-64::xorg-libxdmcp-1.1.3-hb9d3cd8_1 
  xorg-libxext       conda-forge/linux-64::xorg-libxext-1.3.4-h0b41bf4_2 
  xorg-libxrender    conda-forge/linux-64::xorg-libxrender-0.9.11-hd590300_0 
  xorg-renderproto   conda-forge/linux-64::xorg-renderproto-0.11.1-hb9d3cd8_1003 
  xorg-xextproto     conda-forge/linux-64::xorg-xextproto-7.3.0-hb9d3cd8_1004 
  xorg-xorgproto     conda-forge/linux-64::xorg-xorgproto-2024.1-hb9d3cd8_1 
  zlib               conda-forge/linux-64::zlib-1.3.1-h4ab18f5_1
```

Reply:
If we are able to reproduce with `openslide` alone, then think we should [file an issue on that feedstock]( https://github.com/conda-forge/openslide-feedstock/issues/new?assignees=&labels=bug&projects=&template=1-bug-report.yml )

Reply:
> If we are able to reproduce with openslide alone, then think we should file an issue on that feedstock

Let me go a little further first and see if I can narrow it down to one of `openslide`'s dependencies. `openslide` might just happen to be our entrypoint into a problem further up the stack (like with `cairo`).

Reply:
Sounds good. Thanks James! 🙏

[`conda-tree`]( https://github.com/conda-incubator/conda-tree ) may be helpful

Reply:
I was able to trace this further, reported here: https://github.com/conda-forge/xorg-libxrender-feedstock/issues/11

> [conda-tree](https://github.com/conda-incubator/conda-tree) may be helpful

Oh cool! Yeah that helps a lot (sorry I didn't see your comment before posting in `xorg-libxrender`).

```shell
docker run \
    --rm \
    -it rapidsai/ci-conda:latest \
    bash

conda config --set path_conflict warn

conda install \
    --override-channels \
    -c conda-forge \
        openslide

conda-tree whoneeds xorg-xorgproto
# xorg-libx11

conda-tree whoneeds xorg-libx11
# xorg-libxext
# xorg-libxrender
# cairo
```

Reply:
Looks like this issue is not specific to this PR... it's causing build failures on nightly builds of `branch-24.12` as well: https://github.com/rapidsai/cucim/actions/runs/11064821834

Reply:
I just restarted all CI here... the nightly build of this project passed last night (https://github.com/rapidsai/cucim/actions/runs/11099715789), suggesting that maybe the upstream dependency issues have been addressed?

See this explanation: https://github.com/conda-forge/xorg-libxrender-feedstock/issues/11#issuecomment-2380637188

We seem to have been caught in the middle of a migration of the `xorg-*` packages. If CI passes here, that might be a sign that enough of them have now been migrated that these clobbering issues no longer affect the subset pulled in by `cucim` and its dependencies.

Reply:
Versions the `conda-build-cpp` jobs are now getting:

```text
    xorg-libice:               1.1.1-h57736b2_1     conda-forge
    xorg-libsm:                1.2.4-hbac51e1_1     conda-forge
    xorg-libx11:               1.8.9-he755bbd_2     conda-forge
    xorg-libxau:               1.0.11-h86ecc28_1    conda-forge
    xorg-libxdmcp:             1.1.3-h57736b2_2     conda-forge
    xorg-libxext:              1.3.4-h57736b2_3     conda-forge
    xorg-libxrender:           0.9.11-h57736b2_1    conda-forge
    xorg-xorgproto:            2024.1-h86ecc28_1    conda-forge
```

([build link](https://github.com/rapidsai/cucim/actions/runs/11058127444/job/30860603007?pr=785))

Think the key change there is that we're now getting build `_1` of `libxrender`, where previously (in the failing jobs) we'd been getting build `_0` of that same version:

```text
    xorg-libxrender:           0.9.11-hd590300_0    conda-forge
```

And that's probably thanks to https://github.com/conda-forge/xorg-libxrender-feedstock/pull/10 being merged two days ago.